pinot-common/src/main/java/org/apache/pinot/common/request/GroupBy.java

```diff
@@ -687,13 +682,13 @@ public void read(org.apache.thrift.protocol.TProtocol prot, GroupBy struct) thro
       }
       if (incoming.get(2)) {
         {
-          org.apache.thrift.protocol.TList _list75 = new org.apache.thrift.protocol.TList(org.apache.thrift.protocol.TType.STRING, iprot.readI32());
-          struct.expressions = new java.util.ArrayList<java.lang.String>(_list75.size);
-          @org.apache.thrift.annotation.Nullable java.lang.String _elem76;
-          for (int _i77 = 0; _i77 < _list75.size; ++_i77)
+          org.apache.thrift.protocol.TList _list83 = new org.apache.thrift.protocol.TList(org.apache.thrift.protocol.TType.STRING, iprot.readI32());
+          struct.expressions = new java.util.ArrayList<java.lang.String>(_list83.size);
+          @org.apache.thrift.annotation.Nullable java.lang.String _elem84;
+          for (int _i85 = 0; _i85 < _list83.size; ++_i85)
           {
-            _elem76 = iprot.readString();
-            struct.expressions.add(_elem76);
+            _elem84 = iprot.readString();
+            struct.expressions.add(_elem84);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 684:693, 667:676.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.